src/backend/api/types.py

```diff
@@ -84,9 +92,78 @@ def __eq__(self, other: object) -> bool:
 
 
 class Object(BaseModel):
-    Name: str
-    Type: Literal["OBJECT", "FOLDER"]
     Key: str
+    VersionId: Optional[str] = None
     LastModified: Optional[dt] = None
     ETag: Optional[str] = None
+    ObjectLockMode: Optional[ObjectLockModeType] = None
+    ObjectLockRetainUntilDate: Optional[dt] = None
+    ObjectLockLegalHoldStatus: Optional[ObjectLockLegalHoldStatusType] = None
+    Owner: Optional[OwnerTypeDef] = None
+    Name: str
     Size: Optional[int] = None
+    Type: Literal["OBJECT", "FOLDER"] = "OBJECT"
+
+
+class DeletedObject(BaseModel):
+    Key: str
+    VersionId: Optional[str] = None
+    DeleteMarker: bool
+    DeleteMarkerVersionId: Optional[str] = None
+
+
+class ObjectIdentifier(BaseModel):
+    Key: str
+    VersionId: str = ""
+
+
+class ObjectRequest(ObjectIdentifier):
+    pass
+
+
+class ListObjectsRequest(BaseModel):
+    Prefix: str = ""
+    Delimiter: str = "/"
+
+
+class ListObjectVersionsRequest(ListObjectsRequest):
+    pass
+
+
+class ObjectVersion(Object):
+    IsDeleted: bool
+    IsLatest: bool
+
+
+class ObjectLockLegalHold(BaseModel):
+    Status: ObjectLockLegalHoldStatusType
+
+
+class ObjectAttributes(Object, TagSet):
+    pass
+
+
+class SetObjectTaggingRequest(ObjectIdentifier, TagSet):
+    pass
+
+
+class SetObjectLockLegalHoldRequest(ObjectIdentifier):
+    LegalHold: ObjectLockLegalHold
+
+
+class RestoreObjectRequest(ObjectIdentifier):
+    pass
+
+
+class DeleteObjectRequest(ObjectIdentifier):
+    pass
+
+
+class DeleteObjectByPrefixRequest(BaseModel):
+    # The prefix of the objects to delete.
+    # Note, a prefix like `a/b/` will delete all objects starting with
+    # that prefix, whereas `a/b` will only delete this specific object.
+    Prefix: str
+    Delimiter: str = "/"
+    # If `True`, all versions will be deleted, otherwise the latest one.
+    AllVersions: bool = False
```

Comment:
I think these could be docstrings, which would make them show in the openapi interface. _I think_.